LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,352 @@
+#pragma once
+#include <functional>
+
+template <typename Type>
+class TSLList
+{
+private:
+	int Size;
+	struct FNode
+	{
+		Type Element;
+		FNode* Next = nullptr;
+	};
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+public:
+	TSLList()
+	{
+		Size = 0;
+	}
+
+	~TSLList()
+	{
+		Clear();
+	}
+
+	//--------------------------------FIterator
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator*()
+		{
+			return Current->Element;
+		}
+
+		const Type& operator*() const
+		{
+			return Current->Element;
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+		bool operator!=(const FIterator& Other) const
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	FIterator begin()
+	{
+		if(Head)
+			return FIterator(Head);
```

Comment:
This line might not be that necessary, as by default, according to your member initialization at the top of the class, Head should start as nullptr. Therefore, lines 70 and 71 would be doing the same either way.